js/src/momoa.d.ts

```diff
@@ -0,0 +1,113 @@
+
+//-----------------------------------------------------------------------------
+// Options
+//-----------------------------------------------------------------------------
+
+type MomoaMode = "json" | "jsonc";
+
+interface MomoaTokenizeOptions {
+    mode: MomoaMode;
+    ranges: boolean;
+}
+
+interface MomoaParseOptions {
+    mode: MomoaMode;
+    ranges: boolean;
+    tokens: boolean;
+}
+
+//-----------------------------------------------------------------------------
+// Nodes
+//-----------------------------------------------------------------------------
+
+export interface MomoaNode {
+    type: string;
+    loc?: MomoaLocationRange;
+    range?: MomoaRange;
+}
+
+/**
+ * The root node of a JSON document.
+ */
+export interface MomoaDocumentNode extends MomoaNode {
+    body: MomoaNode;
```

Comment:
do Document, Value, Element etc have specific `type` values? It's likely that they can be given literal types like `type: 'document'` here, and then the AnyNode union will be discriminated, and things like `if (node.type === "document")` will turn node into a DocumentNode.

Reply:
Oh nice! Didn't realize I could do that.